observability-lib/grafana/builder.go

```diff
@@ -104,33 +111,39 @@ func (b *Builder) AddNotificationPolicy(notificationPolicies ...*alerting.Notifi
 	b.notificationPoliciesBuilder = append(b.notificationPoliciesBuilder, notificationPolicies...)
 }
 
-func (b *Builder) Build() (*Dashboard, error) {
-	db, errBuildDashboard := b.dashboardBuilder.Build()
-	if errBuildDashboard != nil {
-		return nil, errBuildDashboard
-	}
+func (b *Builder) Build() (*Observability, error) {
+	observability := Observability{}
 
-	var alerts []alerting.Rule
-	for _, alertBuilder := range b.alertsBuilder {
-		alert, errBuildAlert := alertBuilder.Build()
-		if errBuildAlert != nil {
-			return nil, errBuildAlert
+	if b.dashboardBuilder != nil {
+		db, errBuildDashboard := b.dashboardBuilder.Build()
+		if errBuildDashboard != nil {
+			return nil, errBuildDashboard
 		}
+		observability.Dashboard = &db
 
-		// Add common tags to alerts
-		if b.alertsTags != nil && len(b.alertsTags) > 0 {
-			tags := maps.Clone(b.alertsTags)
-			maps.Copy(tags, alert.Labels)
+		var alerts []alerting.Rule
+		for _, alertBuilder := range b.alertsBuilder {
+			alert, errBuildAlert := alertBuilder.Build()
+			if errBuildAlert != nil {
+				return nil, errBuildAlert
+			}
 
-			alertBuildWithTags := alertBuilder.Labels(tags)
-			alertWithTags, errBuildAlertWithTags := alertBuildWithTags.Build()
-			if errBuildAlertWithTags != nil {
-				return nil, errBuildAlertWithTags
+			// Add common tags to alerts
+			if b.alertsTags != nil && len(b.alertsTags) > 0 {
+				tags := maps.Clone(b.alertsTags)
+				maps.Copy(tags, alert.Labels)
```

Comment:
what is the difference between the tags and alert.Labels at this point?

Reply:
Its the same I use tags as a name but grafana calls them Labels I could rename vars

Reply:
non-blocking, but if you could clarify in a future iteration please